rs/interfaces/src/ingress_pool.rs

```diff
@@ -19,28 +19,33 @@ pub struct IngressPoolObject {
     /// The MessageId of the RawIngress message
     pub message_id: MessageId,
 
+    /// Which peer the node has received the ingress message from.
+    pub peer_id: NodeId,
```

Comment:
nit: this is the originator if the ingress message ideally, the replica that received the ingress from the BN.

Maybe it is worth renaming. Also maybe it is worth adding a todo that the originator id should be determined by extracting the signer once we sign ingress messages.

Reply:
Agreed, renamed, and added a TODO.